nautobot/project-static/js/job_result.js

```diff
@@ -44,9 +46,13 @@ $(document).ready(function(){
                     // Update the job logs table
                     updateLogTable(pending_result_id);
 
-                    // If there is a terminal status, refresh the page.
+                    /// Should reload the page yet?
+                    let reload_page = window.sessionStorage.getItem("job_results_current_page") == null;
+
+                    // If there is a terminal status, refresh the page and clear session storage.
                     if (terminal_statuses.includes(data.status.value)) {
-                        window.location.reload();
+                        reload_page && window.location.reload();
+                        window.sessionStorage.removeItem("job_results_current_page");
```

Comment:
```suggestion
                        window.sessionStorage.removeItem("job_results_current_page");
                        reload_page && window.location.reload();
```
Reloading the page would cause us to never hit the removeItem but it's no longer needed if we hit a terminal state anyway.